src/domain/services/query.service.ts

```diff
@@ -63,31 +68,58 @@ export class QueryService {
     return results;
   }
 
-  private addMatches(match: any, qb: QueryBuilder<object>, parentEntity) {
-    const parentId = parentEntity === Product ? 'id': 'product_id';
+  private parseFilter(matches): [string, any][] {
```

Comment:
It would be useful to add some comments with the examples that are in the tests, to understand the expected input. I didn't remember the mongodb syntax.

```
        amino_acids_tags: { $all: [aminoValue, aminoValue2] }
 
        $and: [
          { amino_acids_tags: aminoValue },
          { amino_acids_tags: aminoValue2 },
        ],
```

Reply:
I feel like the comment would be a bit redundant. e.g. "$all means select products where all of the named tags are present". Is there something else you were meaning @stephanegigandet?

Reply:
I've added comments to the main code (as opposed to the tests)